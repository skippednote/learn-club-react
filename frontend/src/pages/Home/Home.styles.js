import styled, { keyframes } from 'react-emotion';

const spin = keyframes`
from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
}`;

export const Logo = styled('div')`
  font-size: 60px;
  color: #39f;
  animation: ${spin} infinite 20s linear;
`;
Logo.displayName = 'Logo';
