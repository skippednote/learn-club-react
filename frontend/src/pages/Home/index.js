// @flow
import React from 'react';
import Counter from '../../components/Counter';
import PeopleList from '../../components/PeopleList';

class Home extends React.Component<{}> {
  render() {
    return (
      <div>
        <h1>Learn Club - React</h1>
        <Counter />
        <PeopleList />
      </div>
    );
  }
}
export default Home;
