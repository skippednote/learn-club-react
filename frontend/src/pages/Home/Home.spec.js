import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Home from './index'

describe('Home', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Home />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('renders the heading', () => {
    const wrapper = shallow(<Home />)
    const heading = wrapper.find('h1').text()
    expect(heading).toEqual('Learn Club - React')
  })

  it('renders Counter component', () => {
    const wrapper = shallow(<Home />)
    const counter = wrapper.find('Counter')
    expect(counter).toHaveLength(1)
  })
})
