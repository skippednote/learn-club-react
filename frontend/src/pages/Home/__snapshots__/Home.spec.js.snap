// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<div>
  <h1>
    Learn Club - React
  </h1>
  <Counter />
  <PeopleList />
</div>
`;
